mysql/Dockerfile.mariadb

```
@@ -1,5 +1,5 @@
 FROM mariadb
 
-COPY docker-healthcheck /usr/local/bin/
+ENV MARIADB_MYSQL_LOCALHOST_USER=1
 
-HEALTHCHECK CMD ["docker-healthcheck"]
+HEALTHCHECK CMD ["/usr/local/bin/healthcheck.sh", "--su=mysql", "--innodb_initialized"]
```

```suggestion
HEALTHCHECK CMD ["healthcheck.sh", "--su=mysql", "--innodb_initialized"]
```

Does it always use InnoDB by default?  (Will this check still be meaningful/useful if InnoDB _isn't_ used?)

---

Technically I can't start without innodb at the moment;
```
$ podman run --rm -e MARIADB_ROOT_PASSWORD=bob  mariadb:10.5 --skip-innodb
2022-03-09 22:42:06+00:00 [Note] [Entrypoint]: Entrypoint script for MariaDB Server 1:10.5.15+maria~focal started.
2022-03-09 22:42:07+00:00 [Note] [Entrypoint]: Switching to dedicated user 'mysql'
2022-03-09 22:42:07+00:00 [Note] [Entrypoint]: Entrypoint script for MariaDB Server 1:10.5.15+maria~focal started.
2022-03-09 22:42:07+00:00 [Note] [Entrypoint]: Initializing database files
2022-03-09 22:42:07 0 [ERROR] Unknown/unsupported storage engine: InnoDB
2022-03-09 22:42:07 0 [ERROR] Aborting
```
If `--innodb_initialized` wasn't there another test from the script would need to be used.

---

Fair -- I guess my question was more "is there a more generic test we could use" but it sounds like this is probably the best one.

My next concern is that the script appears to use different `mysql` options depending on the test, so some might use the unix socket, right?  So this will potentially be reporting the container as "healthy" even while the initialization scripts are still running, especially if they're taking a long time (like a really large import)?

Should we always start with `--connect` (that uses `--protocol tcp` explicitly, presumably to combat this issue)?

---

You are right in that `innodb_initialized` and `mariadbupgrade` are the ones that could report healthy on their own ahead of fully started. I'll get that documented.